### PR TITLE
Warning for photomet added

### LINKED ISSUE #3451: Photomet topo mode allows user to select DEM source
**ISIS version(s) affected**: x.y.z
Was an issue in 2017.  I am not a technical user and can no longer run ISIS3 to verify it is still an issue.

**Description**  

When setting normname=topo, the algorithm must be applied to an ellipsoid shape model so a DEM can be calculated.  However, photomet allows the user to select anglesource=DEM.  

**How to reproduce**  

Open photomet gui.
Select normname=topo
Select anglesource=DEM
Observe that no error/warning given to the user.  
Can do the same from the command line if you prefer.  

**Possible Solution**  

Remove DEM as a valid value for the anglesource parameter when normname=topo.  
At least throw a warning to the user that the selected parameters are mutually incompatible and one of the who needs to be change.  

**Additional context**

### PR DESCRIPTION
**Pull request by ladoramkershner**
Added warning if 'DEM' angle source option is used with 'mixed' or 'topo' normalization method

## Description
A warning is issued to the application logs but still outputs result cube. This warning style was chosen because some information as to why the angle source/normalization methods cannot be paired is not clearly outlined. We do not want to prevent users who understand the fundamentals behind the photomet methods from making an educated decision to couple 'DEM' and 'Mixed'/'Topo'.

## Related Issue
Closes #3451 and #3452. 

## Motivation and Context
There is some lost information within the building as to why this change is necessary, however, it is based on tickets #3451 and #3452, which were migrated from redmine. 

## How Has This Been Tested?
New tests for photomet were added to test that the new warning is appropriately output. 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by krlberry — DISMISSED** [review]
:+1: Looks good!

 I'm honestly not sure if we're still doing this or not, but in the past, for changes to ISIS applications, we would add a history entry which explains the change that was made to the documentation xml file associated with the ISIS application. (In this case, `photomet.xml`) I have an example of this in my PR for `voy2isis` #3490.

I'm approving the PR anyway since I'm not sure whether this is necessary or not.

**Comment by jessemapel** [comment]
@kberryUSGS That is still our standards

**Comment by Kelvinrr** [comment]
Looks good to me, will approve after history comment is added.

**Comment by krlberry** [comment]
@jessemapel :+1: Thanks. It had been so long since I'd actually updated an ISIS application that I couldn't remember.

File: isis/src/base/apps/photomet/main.cpp [hunk 1]
```diff
@@ -1758,6 +1758,12 @@ void IsisMain() {
   // determine how photometric angles should be calculated
   angleSource = ui.GetString("ANGLESOURCE");
 
+  if ((normName == "TOPO" || normName == "MIXED") && angleSource == "DEM") {
+    QString message = "The " + normName + " Normalized model is not recommended for use with the " + angleSource + " Angle Source option";
+    PvlGroup warning("Warning");
+    warning.addKeyword(PvlKeyword("Warning",message),Pvl::Replace);
+    Application::Log(warning);
+  }
```
Comment (scsides) [inline isis/src/base/apps/photomet/main.cpp:1766]: Using Pvl::Replace will overwrite any other "Warning" keywords in a group. Yes the group was just created, so why add the Replace at all? If additional tests similar to this one are added they should add an additional "Warning" keyword to this group. With the "Replace" only one will show in the log.

File: isis/src/base/apps/photomet/tsts/testDemWarning/Makefile [hunk 2]
```diff
@@ -0,0 +1,31 @@
+APPNAME = photomet
+
+angleSourceDemUsedemFalse.cub.TOLERANCE = 0.001
+angleSourceDemUsedemTrue.cub.TOLERANCE = 0.001
```
Comment (scsides) [inline isis/src/base/apps/photomet/tsts/testDemWarning/Makefile]: These look like leftovers from a cut-n-paste. Should be removed

**Review by scsides — CHANGES_REQUESTED** [review]

File: isis/src/base/apps/photomet/main.cpp (hunk 1 above)
Comment (scsides) [reply isis/src/base/apps/photomet/main.cpp:1766]: Suggest changing the group name to "Warnings"

**Comment by ladoramkershner** [comment]
>  Looks good!
> 
> I'm honestly not sure if we're still doing this or not, but in the past, for changes to ISIS applications, we would add a history entry which explains the change that was made to the documentation xml file associated with the ISIS application. (In this case, `photomet.xml`) I have an example of this in my PR for `voy2isis` #3490.
> 
> I'm approving the PR anyway since I'm not sure whether this is necessary or not.

@kberryUSGS @jessemapel The entirety of photomet.xml is getting reported in the file change section of this PR. So to hopefully save some sanity, in response to your comments I added:
```
<change name="Lauren Adoram-Kershner" date="2019-11-07">
    Added a warning when the 'DEM' angle source option is used with 'mixed'
    or 'topo' normalization method. Fixes #3451 and #3452.
</change>
```
at the end of the <history> section. I know this is vague, but because there was not a lot of explanatory information surrounding the request for this change, I was not sure what to put. If you have any suggestions, I am willing to add more.

File: isis/src/base/apps/photomet/main.cpp (hunk 1 above)
Comment (ladoramkershner) [reply isis/src/base/apps/photomet/main.cpp:1766]: I changed the pvlgroup name to Warnings and removed the Pvl::Replace.

**Review by jessemapel — APPROVED** [review]

**Comment by jessemapel** [comment]
> @kberryUSGS @jessemapel The entirety of photomet.xml is getting reported in the file change section of this PR.

If you click on the little gear you can hide whitespace changes like that. It has to do with slick edit adding spaces and atom removing them.

**Review by krlberry — APPROVED** [review]

**Review by scsides — APPROVED** [review]
Looks good

**Pull request merged**